src/DataLayouts/DataLayouts.jl

```diff
@@ -411,6 +415,100 @@ end
     IFH{SS, Ni}(dataview)
 end
 
+# ======================
+# Data0D DataLayout
+# ======================
+
+Base.size(data::Data0D) = (1, 1, 1, 1, 1)
+
+struct DataF{S, A} <: Data0D{S}
+    array::A
+end
+
+function DataF{S}(array::AbstractArray{T, 2}) where {S, T}
+    @assert size(array, 1) == 1
+    check_basetype(T, S)
+    DataF{S, typeof(array)}(array)
+end
+
+function DataF{S}(array::AbstractVector{T}) where {S, T}
+    @assert size(array, 1) == 1
+    @assert typesize(T, S) == 1
+    DataF{S}(reshape(array, (:, 1)))
+end
+
+function DataF{S}(ArrayType) where {S}
+    T = eltype(ArrayType)
+    DataF{S}(ArrayType(undef, 1, typesize(T, S)))
+end
+
+# @inline function Base.getproperty(data::VF{S}, i::Integer) where {S}
+#     array = parent(data)
+#     T = eltype(array)
+#     SS = fieldtype(S, i)
+#     offset = fieldtypeoffset(T, S, i)
+#     nbytes = typesize(T, SS)
+#     dataview = @inbounds view(array, :, (offset + 1):(offset + nbytes))
+#     VF{SS}(dataview)
+# end
+@inline function Base.getproperty(data::DataF{S}, i::Integer) where {S}
+    array = parent(data)
+    T = eltype(array)
+    SS = fieldtype(S, i)
+    offset = fieldtypeoffset(T, S, i)
+    nbytes = typesize(T, SS)
+    dataview = @inbounds view(array, :, (offset + 1):(offset + nbytes))
+    DataF{SS}(dataview)
+end
+
+# @inline function Base.getindex(data::VF{S}, v::Integer) where {S}
+#     @boundscheck 1 <= v <= size(parent(data), 1) ||
+#                  throw(BoundsError(data, (v,)))
+#     dataview = @inbounds view(parent(data), v, :)
+#     @inbounds get_struct(dataview, S)
+# end
+@inline function Base.getindex(data::DataF{S}, v::Integer) where {S}
+    @boundscheck 1 <= v <= size(parent(data), 1) ||
+                 throw(BoundsError(data, (v,)))
+    dataview = @inbounds view(parent(data), v, :)
+    @inbounds get_struct(dataview, S)
+end
+
+# @propagate_inbounds function Base.getindex(
+#     col::DataColumn,
+#     I::CartesianIndex{5},
+# )
+#     col[I[4]]
+# end
+@propagate_inbounds function Base.getindex(
+    col::Data0D,
+    I::CartesianIndex{5},
+)
+    col[I[4]]
+end
```

Comment:
@simonbyrne can you tell me about these methods for get & set index? This is a place where I more or less have blindly copied.